src/holon/views/holon.py

```diff
@@ -49,13 +51,17 @@ def post(self, request: Request):
                         )
 
                 HolonV2Service.logger.log_print(f"Cloning scenario {data['scenario'].id}")
-                scenario = rule_mapping.get_scenario_and_apply_rules(
-                    data["scenario"].id, data["interactive_elements"]
+                cloned_scenario = rule_mapping.apply_rules(
+                    original_scenario, data["interactive_elements"]
                 )
 
+                # prefetch again after rules are applies, for more efficient serialization
+                cloned_scenario = Scenario.queryset_with_relations().get(id=cloned_scenario.id)
+                scenario_to_delete = cloned_scenario
```

Comment:
During development a few times I've accidentally removed the wrong scenario or the wrong (stale) relations of the new scenario.

I wanted it to be more explicit that the scenario will be removed, so that the developer only schedules the removal when the scenario is in the correct state to be removed. By putting it in a variable specifically for that purpose.

Reply:
I actually thought naming the cloned scenario `cloned_scenario` made a lot of sense. I was just confused because while reading through the code having a second object `scenario_to_delete` makes it seem like there are two different scenarios, and `cloned_scenario` is not the one being deleted. 

Therefore I'd say keep the change where cloned scenarios are called `cloned_scenario`, but dont make a second variable `scenario_to_delete` and instead just delete `cloned_scenario`.

Reply:
The issue is that on line 55 the cloned_scenario is not ready to be deleted, that would delete some relations of the original scenario.

We'll discuss later, I guess I'm ok with the renaming.

Reply:
Nvm I think I'm wrong about the thing on line 55 but we should verify it anyway.